xrpl/models/transactions/account_delete.py

```diff
@@ -42,3 +44,22 @@ class AccountDelete(Transaction):
         default=TransactionType.ACCOUNT_DELETE,
         init=False,
     )
+
+    credential_ids: Optional[Set[str]] = None
```

Comment:
This should be a list, not a set.

Reply:
I used a `Set` in order to de-deduplicate the inputs, if any

Reply:
A set might break some of the assumptions in the library, I would still stick with a list.

Reply:
> Do you have a suspicion on any specific part of the library? I can dig deeper to see if something is amiss.

For example, `from_dict` and `to_dict`, and maybe certain parts of the binary codec. I don't want to add additional complexity to this PR, so I don't think it's worth digging into right now.

> Is it alright if the client library doesn't de-deduplicate the input?

Yes.

> There are only two unique credentials in this input, but I won't be able to perform any length checks at all, due to the lack of de-deduplication.

rippled uses the total length, not the de-duped length. Client libraries should do the same.